src/controller/CHIPDeviceController.cpp

```diff
@@ -2175,6 +2178,44 @@ void DeviceCommissioner::ParseFabrics()
     CommissioningStageComplete(return_err, report);
 }
 
+void DeviceCommissioner::ParseIcdInfo()
+{
+    CHIP_ERROR err;
+    IcdInfo info;
+    IcdManagement::Attributes::FeatureMap::TypeInfo::DecodableType featureMap;
+
+    // TODO(#29382): We probably want to read "ActiveMode" attribute (to be implemented) for ICD.
+    err = mAttributeCache->Get<IcdManagement::Attributes::FeatureMap::TypeInfo>(kRootEndpointId, featureMap);
+    if (err == CHIP_NO_ERROR)
+    {
+        info.isIcd                  = true;
+        info.checkInProtocolSupport = !!(featureMap & 0x1);
+    }
+    else if (err == CHIP_ERROR_IM_STATUS_CODE_RECEIVED)
+    {
+        app::StatusIB statusIB;
+        err = mAttributeCache->GetStatus(
+            app::ConcreteAttributePath(kRootEndpointId, IcdManagement::Id, IcdManagement::Attributes::FeatureMap::Id), statusIB);
+        if (err != CHIP_NO_ERROR)
+        {
+            CommissioningStageComplete(err);
```

Comment:
So why is error reading from the cache leading to immediate CommissioningStageComplete with no report, while error returned by the server leads to a report, but with uninitialized info?

Reply:
No longer appliable